qpid-jms-client/src/main/java/org/apache/qpid/jms/provider/amqp/AmqpSaslAuthenticator.java

```diff
@@ -137,7 +142,9 @@ private void handleSaslStep() throws JMSSecurityException {
                 byte[] challenge = new byte[sasl.pending()];
                 sasl.recv(challenge, 0, challenge.length);
                 byte[] response = mechanism.getChallengeResponse(challenge);
-                sasl.send(response, 0, response.length);
+                if (response != null) {
```

Comment:
Thanks for the detailed feedback.
will need to check if that can go, but I think not b/c it delegates to the saslclient which can return null. 
javax.security.sasl.SaslClient#evaluateChallenge